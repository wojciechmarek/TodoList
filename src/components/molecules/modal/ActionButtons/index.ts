export * from './action-buttons';
export * from './action-buttons.interface';
