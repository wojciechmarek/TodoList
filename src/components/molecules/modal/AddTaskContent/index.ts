export * from './add-task';
export * from './add-task.interface';
