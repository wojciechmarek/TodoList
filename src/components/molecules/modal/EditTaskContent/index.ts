export * from './edit-task';
export * from './edit-task.interface';
