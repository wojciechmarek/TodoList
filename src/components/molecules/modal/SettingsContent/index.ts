export * from './settings-content';
export * from './settings-content.interface';
