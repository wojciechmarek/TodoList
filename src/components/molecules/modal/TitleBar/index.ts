export * from './title-bar';
export * from './title-bar.interface';
