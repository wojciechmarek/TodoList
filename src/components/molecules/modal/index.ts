export * from './TitleBar';
export * from './SettingsContent';
export * from './EditTaskContent';
export * from './ActionButtons';
