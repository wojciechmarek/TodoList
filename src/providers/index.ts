export * from './Modal';
export * from './Common';
export * from './Theme';
export * from './Storage';
