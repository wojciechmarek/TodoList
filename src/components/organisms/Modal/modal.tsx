import styled from '@emotion/styled';
import { useModalProvider } from '../../../providers';

const ModalStyle = styled.div<{ visible: boolean }>`
  z-index: 1;
  display: ${({ visible }) => (visible ? 'flex' : 'none')};
  position: absolute;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
  background-color: rgba(0, 0, 0, 0.4);
`;

export const Modal = () => {
  const { isOpen, content } = useModalProvider();

  return <ModalStyle visible={isOpen}>{content}</ModalStyle>;
};
