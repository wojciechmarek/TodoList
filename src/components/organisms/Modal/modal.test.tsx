import { configure, render } from '@testing-library/react';
import { Modal } from './modal';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import { ModalProvider } from '../../../providers';

describe('Modal', () => {
  beforeEach(() => {
    configure({ throwSuggestions: true });
  });

  it('should render', () => {
    const { baseElement } = render(
      <ModalProvider>
        <Modal />
      </ModalProvider>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(
      <ModalProvider>
        <Modal />
      </ModalProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
