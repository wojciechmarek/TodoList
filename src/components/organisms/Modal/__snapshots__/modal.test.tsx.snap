// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snapshot 1`] = `
.emotion-0 {
  z-index: 1;
  display: none;
  position: absolute;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
  background-color: rgba(0, 0, 0, 0.4);
}

<div
  className="emotion-0"
/>
`;
